substrate/frame/support/procedural/src/benchmark.rs

```diff
@@ -762,7 +767,7 @@ fn expand_benchmark(
 				Expr::Cast(t) => {
 					let ty = t.ty.clone();
 					quote! {
-						<<T as frame_system::Config>::RuntimeOrigin as From<#ty>>::from(#origin);
```

Comment:
all of this seem to have been hardcoded 🙈 well, not like anything broke because of it, but indeed not a good practice.

Reply:
unless if was intentional? but I can't see a reason.

Reply:
I can't see a reason either, and we need this unhardcoded moving forward to work with the frame umbrella crate